test/profiler/test_profiler.py

```diff
@@ -883,18 +895,19 @@ def join_threads(context: bool):
         ]
```

Comment:
is line 875 - 888 supposed to wait for all threads to clean up first? Is that logic not working properly for et?